getting_started/scripting/gdscript/gdscript_styleguide.rst

```diff
@@ -688,36 +688,42 @@ Static typing
 
 Since Godot 3.1, GDScript supports :ref:`optional static typing<doc_gdscript_static_typing>`.
 
-Type hints
-~~~~~~~~~~
+Declared Types
+~~~~~~~~~~~~~~
 
-Place the colon right after the variable's name, without a space, and let the
-GDScript compiler infer the variable's type when possible.
+To declare a variable's type, use ``<variable>: <type>``:
 
-**Good**:
+::
+
+   var health: int = 0
+
+To declare the return type of a function, use ``-> <type>``: 
 
 ::
 
-   onready var health_bar: ProgressBar = get_node("UI/LifeBar")
+   func heal(amount: int) -> void:
 
-   var health := 0 # The compiler will use the int type.
+Inferred Types
+~~~~~~~~~~~~~~
 
-**Bad**:
+In some cases you can let the compiler infer the type, using ``:=``:
```

Comment:
I'd say the idea is to use type inference whenever you can. It makes your code easier to change (you don't have to change the type when refactoring) and works in many cases. The exception is when it doesn't work, i.e. on get_node() calls, because unless the scene or file of the node is loaded in memory, the compiler lacks context to infer the type.